rover/src/main.c

```diff
@@ -305,20 +327,22 @@ void read_us()
 		us_br = get_us(TRIGBR, ECHOBR, gpioc);
 		us_bc = get_us(TRIGBC, ECHOBC, gpiob);
 		us_bl = get_us(TRIGBL, ECHOBL, gpiob);
-		k_sleep(DELAY);
+		k_yield();
 	}
 }
 
 void uart_out()
 {
-		printk("Front Left US: %d\tFront Center US: %d\tFront Right US: %d\nBack Right US: %d\tBack Center US: %d\tBack Letf US: %d\nFront Right IR: %d\tFront Left IR: %d\nBack Right IR: %d\tBack Letf IR: %d\nDirection: %s\n", us_fl, us_fc, us_fr, us_br, us_bc, us_bl, irfr, irfl, irbr, irbl, tdir);
+	printk("Front Left US: %d\tFront Center US: %d\tFront Right US: %d\nBack Right US: %d\tBack Center US: %d\tBack Letf US: %d\nFront Right IR: %d\tFront Left IR: %d\nBack Right IR: %d\tBack Letf IR: %d\nDirection: %s\n", us_fl, us_fc, us_fr, us_br, us_bc, us_bl, irfr, irfl, irbr, irbl, tdir);
```

Comment:
Do we need debug out here? Will the rover be connected to Serial while running?

Reply:
I just left the function for easy debug, just call it in any loop.

Reply:
I can see in few places it is being used. I would suggest you to keep the function but not use it anywhere in the application. Also, the function name can be changed to `debug_out` and add a comment as below;

`This function can be used to debug the application. Call it anywhere in the threads to see the Rover parameters over serial.`